test/runnable/ufcs.d

```diff
@@ -128,8 +128,8 @@ void test3()
     assert(1    .init == 0);
     assert([1]  .init == null);
     assert([1:1].init == null);
-    assert(1.0  .init is double.nan);
-    assert(10i  .init is idouble.nan);
+    assert(1.0  .init is double.init);
```

Comment:
why change this?

Reply:
Init and nan have different bit patterns. I imagine was the reason why the original patch changed this.

Reply:
They should have the same bit patterns. We tried to make .init a signalling nan, but this was a complete failure. .init and .nan should both be a quiet nan.

Reply:
Is there an issue for this? Also detailing the why and how it is a failure. (I've never been involved in any discussion that made that claim).

Sounds like such a language change should be a prerequisite for this anyhow, and not made here.

Reply:
Because a signalling nan converts to a quiet nan when used. The problem was nobody could come up with a consistent definition of "used".

Reply:
>They should have the same bit patterns.

They don't
```d
double a, b = double.nan;
    writeln(*cast(ulong*)&a, "\n", *cast(ulong*)&b);
```
prints
```
9219994337134247936
9221120237041090560
```

Reply:
They currently don't. They should though.

I take this to mean that the signalling nan init should be deprecated somehow.

Reply:
Ah, right.

Reply:
> is there an issue for this?

There is now: https://issues.dlang.org/show_bug.cgi?id=19905

Reply:
This is fixed.